src/remote-control/process-update-delegate.js

```diff
@@ -132,6 +148,50 @@ export default class ProcessUpdateDelegate {
         return Promise.resolve(ack);
     }
 
+    /**
+     * Callback to invoke when receiving a message as an iq.
+     *
+     * @param {Object} iq - The iq representing the message.
+     * @returns {Promise} Resolves with an ack as an iq.
+     */
+    onMessage(iq) {
+        const from = iq.getAttribute('from');
+        const message = iq.getElementsByTagName('message')[0];
+        const messageType = message.getAttribute('type');
+        const data = JSON.parse(message.textContent);
+        const ack = $iq({
+            id: iq.getAttribute('id'),
+            to: from,
+            type: 'result'
+        });
+
+        switch (messageType) {
+        case MESSAGES.JITSI_MEET_UPDATE:
+            // It is currently assumed all messages received here are for the
+            // remote control, are from the Jitsi-Meet participant, and are all
+            // related to screensharing.
+            this._screenshareConnection
+                && this._screenshareConnection.processMessage({
+                    data,
+                    from
+                });
+
+            break;
+
+        case MESSAGES.REMOTE_CONTROL_UPDATE:
+            // Spot receives messages from remote controls to pass to
+            // Jitsi-Meet.
+            this._sendEventIfInMeeting({
+                data,
+                from
+            });
+
+            break;
+        }
+
+        return Promise.resolve(ack);
```

Comment:
we return a promise here just to make it easier for the caller to chain some things together?  nothing actually async here (that the ack is based on, at least) right?

Reply:
Correct, this is for caller chaining, but you are right that there isn't a need to do anything async. What I bet happened is at some point I wanted to do promise.reject but thinking now that wouldn't make sense because the rejection/error would be set during ack creation time. I should probably audit the promise use here.